### PR TITLE
feat: upgrade wagmi + rainbowkit

### PR DESCRIPTION
**Pull request by janndriessen**
## **Summary of Changes**

* Upgrade RainbowKit and wagmi library
* Adds viem (as required dependency)
* Updates a few places according to library version changes
* Temporarily removes ledger connect support (as required wagmi version is too old) ⚠️ 

###### _By submitting this pull request, you are confirming the following to be true:_

- I have reviewed the [Contribution Guidelines](https://github.com/IndexCoop/index-app/blob/master/CONTRIBUTING.md).
- I have performed a self-review of my own code.
- I have updated my repository to match the master at `IndexCoop/index-app`.
- I have included test data or screenshots that prove my fix is effective or that my feature works.

### CODE REVIEW

File: src/lib/hooks/useBestQuote/index.ts [hunk 1]
```diff
@@ -53,7 +54,7 @@ export interface EnhancedFlashMintQuote extends Quote {
 
 export interface ZeroExQuote extends Quote {
   chainId: string
-  data: string
+  data: Hex
```
Comment (janndriessen) [inline src/lib/hooks/useBestQuote/index.ts:57]: Has to be `Hex` so wagmi accepts `data` as input for sending a tx.

File: src/lib/utils/wagmi.ts [hunk 2]
```diff
@@ -30,35 +30,37 @@ export const ledgerConnector = new IFrameEthereumConnector({
...
       wallets: [
-        walletConnectWallet({ chains }),
+        walletConnectWallet({ chains, projectId }),
         braveWallet({ chains }),
-        trustWallet({ chains }),
+        trustWallet({ chains, projectId }),
       ],
     },
-  ])().concat([ledgerConnector])
+  ])() // .concat([ledgerConnector])
```
Comment (janndriessen) [inline src/lib/utils/wagmi.ts:60]: We currently can't support the ledger connector lib - as it's on an old version of wagmi.

Created an issue to ask for support: https://github.com/LedgerHQ/ledger-live-wagmi-connector/issues/28

File: src/lib/hooks/useTrade.ts [hunk 3]
```diff
@@ -64,29 +34,17 @@ export const useTrade = () => {
...
-          recklesslySetUnpreparedRequest: req,
+        const request = await prepareSendTransaction({
+          account: address,
+          chainId: Number(quote.chainId),
+          to: quote.to,
+          data: quote.data,
+          value: BigInt(quote.value ?? 0),
         })
+        const { hash } = await sendTransaction(request)
+        console.log(hash)
```
Comment (janndriessen) [inline src/lib/hooks/useTrade.ts:47]: Can be done way simpler now

Comment (0xonramp) [reply src/lib/hooks/useTrade.ts:47]: Also no longer 'reckless' 🔥

**Review by 0xonramp — APPROVED** [review]
LGTM 👍

File: src/lib/utils/wagmi.ts (hunk 2 above)
Comment (ckoopmann) [reply src/lib/utils/wagmi.ts:60]: That repo is deprecated so I guess they wont be upgrading to the new wagmi version.

**Pull request merged**